blockchain/chain_sync/src/bucket.rs

```diff
@@ -34,29 +35,50 @@ impl SyncBucket {
 
         false
     }
-    fn add(&mut self, ts: Arc<Tipset>) {
+    /// Adds tipset to vector to be included in the bucket
+    pub fn add(&mut self, ts: Arc<Tipset>) {
         if !self.tips.iter().any(|t| *t == ts) {
             self.tips.push(ts);
         }
     }
+    /// Returns true if SyncBucket is empty
+    pub fn is_empty(&self) -> bool {
+        self.tips.is_empty()
+    }
 }
 
 /// Set of tipset buckets
-#[derive(Default)]
+#[derive(Default, Clone)]
 pub(crate) struct SyncBucketSet {
     buckets: Vec<SyncBucket>,
 }
 
 impl SyncBucketSet {
+    /// Inserts a tipset into a bucket
     pub(crate) fn insert(&mut self, tipset: Arc<Tipset>) {
         for b in self.buckets.iter_mut() {
-            if b.same_chain_as(&tipset) {
+            if b.same_chain_as(tipset.clone()) {
```

Comment:
this clone is required.

Reply:
Oh, didn't see the change in the same_chain_as function, why is that function signature changed to arc as a parameter? you should only need the reference to the TS

Reply:
In your previous review, you mentioned that these operations within `bucket.rs` are now cloning full tipsets instead of arc references....

https://github.com/ChainSafe/forest/pull/293#discussion_r402426818

Reply:
That comment was referring to the change of the underlying type and how you were cloning the whole tipset when the change was made, for functions like this there is no reason to have to pass an Arc<Tipset> if you only need the reference to that tipset for a check (basically just keeping it how it was before). Sorry if I worded poorly